archive/p/python/game-of-life.py

```diff
@@ -1,108 +1,120 @@
-
-    def _populate(self):
-        self.grid = [[self._create_cell() for _ in range(self.width)] for _ in range(self.width)]
-
-    def _link(self):
-        for curr_row, row_list in enumerate(self.grid):
-            previous_row = (curr_row - 1) % self.width
-            next_row = (curr_row + 1) % self.width
-            for curr_col, cell in enumerate(row_list):
-                previous_col = (curr_col - 1) % self.width
-                next_col = (curr_col + 1) % self.width
-                cell.neighbors.append(self.grid[curr_row][previous_col])
-                cell.neighbors.append(self.grid[curr_row][next_col])
-                cell.neighbors.append(self.grid[previous_row][curr_col])
-                cell.neighbors.append(self.grid[next_row][curr_col])
-                cell.neighbors.append(self.grid[previous_row][previous_col])
-                cell.neighbors.append(self.grid[next_row][next_col])
-                cell.neighbors.append(self.grid[previous_row][next_col])
-                cell.neighbors.append(self.grid[next_row][previous_col])
-                cell.grid(column=curr_col, row=curr_row)
-
-    def generate(self):
-        self._populate()
-        self._link()
-        self.after(self.delay, self.step)
-
-    def step(self):
-        for row in self.grid:
-            for cell in row:
-                cell.transition()
-                cell['bg'] = cell.get_background_color()
-        for row in self.grid:
-            for cell in row:
-                cell.clear_state()
-        self.frame_index += 1
-        if self.frame_index < self.total_frames:
-            self.after(self.delay, self.step)
-
-
-def main():
-    if len(sys.argv) > 4:
-        width = int(sys.argv[1])
-        frame_rate = float(sys.argv[2])
-        total_frames = int(sys.argv[3])
-        spawn_rate = float(sys.argv[4])
-        my_grid = Grid(width, spawn_rate, frame_rate, total_frames)
-    else:
-        my_grid = Grid()
-    my_grid.generate()
-    my_grid.mainloop()
-
-
-if __name__ == '__main__':
-    main()
+root = Tk()
+root.title("The Renegade Coder's Game of Life")
+frame = Frame(root, width=720, height=720)
+frame.pack()
+canvas = Canvas(frame, width=720, height=720)
+canvas.pack()
+
+def create_grid():
+	x = 10
+	y = 10
+	global grid # Variable to store the Cell objects
+	global rectangles # Variable to store rectangles
+	rectangles = []
+	grid = []
+	for i in range(70):
+		grid.append([])
+		rectangles.append([])
+		for j in range(70):
+			rect = canvas.create_rectangle(x, y, x+10, y+10, fill="white")
+			rectangles[i].append(rect)
+			grid[i].append(Cell(x, y, i, j))
+			x += 10
+		x = 10
+		y += 10
+
+def rectangle_coordinates(x, y):
+	return (x- x%10, y - y%10)
+
+def click_colour_change(event):
+	print (event.x, event.y)
+	x, y = rectangle_coordinates(event.x, event.y)
+	try:
+		iy = x / 10 - 1
+		ix = y / 10 - 1
+		if ix == -1 or iy == -1:
+			raise IndexError
+			if grid[ix][iy].isAlive:
```

Comment:
I opened up the code in PyCharm quickly, and I noticed there's an error here. These if statements are unreachable due to the raising of the IndexError.